.github/workflows/add-label-when-promoted.yaml

```diff
@@ -0,0 +1,36 @@
+name: Check if commits are promoted
+
+on: [push]
+
+jobs:
+  check-commit:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for all tags and branches
+
+      - name: Label PRs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Get the list of the latest commits
+          commits=$(git log --format=%B -n 100 HEAD~100..HEAD)
+
+          echo "$commits" | while read line ; do
+            if [[ $line =~ Closes\ .*#([0-9]+) ]]; then
+              pr_number=${BASH_REMATCH[1]}
```

Comment:
Closes will be added automatically by the maintainers since they close the PR and not merge, they are using `./script/pull_github_pr.sh <number of PR>`  for adding this info to the commit.

As for your second comment, the label is in the same way as PR. So in case we have a commit with multiple commits, it will be added only once time, see an example for backport PR with multiple commits
![image](https://github.com/scylladb/scylladb/assets/9914527/418dbac7-ca3d-4dc8-8319-ae740a7337e5)